### PR TITLE
Handle artifact directory name collisions

### PR DESCRIPTION
**Pull request by agaleana**
## Summary
- guard the loop that strips redundant artifact directory prefixes to avoid rejecting valid filenames and prevent infinite iteration
- allow saving files whose names match the artifacts directory after removing redundant prefixes
- add regression coverage for saving `artifacts/artifacts`

## Testing
- PYTHONPATH=. pytest tests/test_utils.py::test_save_artifact_strips_redundant_folder tests/test_utils.py::test_save_artifact_allows_filename_matching_directory -q

------
https://chatgpt.com/codex/tasks/task_e_68d07bb6c89c8332b5fbfb734ce9a0fc

### CODE REVIEW

File: utils/artifacts.py [hunk 1]
```diff
@@ -103,8 +103,31 @@ def resolve_artifact_path(
...
+                    except ValueError:
+                        break
+                    if relative_target == Path('.'):
+                        if not stripped_prefix:
+                            raise ArtifactError(
+                                "Filename must not resolve to the artifacts directory itself."
+                            )
+                        break
+                    cleaned_target = relative_target
+                    stripped_prefix = True
```
[the remark this answers is not in the document]
Comment (agaleana) [reply utils/artifacts.py:124]: I have successfully addressed the Gemini code review feedback and fixed the critical bug! Here's what was accomplished:

Issues Fixed:

1. Critical Bug in Prefix Stripping Logic:
◦  Problem: The stripped_prefix flag was incorrectly preventing files named "artifacts" from being saved in the artifacts directory
◦  Solution: Removed the stripped_prefix flag and simplified the while loop as suggested by Gemini
◦  Result: Files can now be saved with names matching the artifacts directory name

Changes Made:

1. utils/artifacts.py:
python
2. tests/test_utils.py:
◦  Added test_save_artifact_with_conflicting_name() to test the specific edge case
◦  This test ensures files named "artifacts" can be saved in an artifacts directory

Verification:

•  ✅ New test passes: test_save_artifact_with_conflicting_name works correctly
•  ✅ All existing tests still pass: 27/27 tests pass (up from 26 before)
•  ✅ No regressions: All artifact-related functionality continues to work
•  ✅ Bug fixed: Files can now be named "artifacts" without causing errors

The solution is more elegant, simpler to understand, and fixes the edge case bug that Gemini identified. The code now properly handles all scenarios including:
•  Redundant prefix stripping (e.g., artifacts/file.txt → file.txt)
•  Files with names matching the directory (e.g., artifacts file in artifacts directory)
•  Path validation and security checks
